### PR TITLE
Fixed issue where map still had focus when instruction list was visib…

### LINKED ISSUE #1009: Map gets the focus when swiping the instructions list
The map it’s getting the focus even if we’re showing the instructions list. If you swipe when the list is shown we’re actually swiping the map and that’s why when you close the list
the `RECENTER` button is shown on top of the `SummaryBottomSheet` 👀 

instructions_list_focus_recenter

### PR DESCRIPTION
**Pull request by devotaaabel**
…le. Also fixed issue where when closing the instruction view, the recenter button was always shown. Fixes #1009

### CODE REVIEW

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationPresenter.java [hunk 1]
```diff
@@ -69,9 +70,12 @@ void onNavigationLocationUpdate(Location location) {
 
   public void onInstructionListVisibilityChanged(boolean visible) {
     if (visible) {
+      recenterButtonVisible = view.isRecenterButtonVisible();
       view.hideRecenterBtn();
     } else {
-      view.showRecenterBtn();
+      if (recenterButtonVisible) {
```
Comment (danesfeder) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationPresenter.java]: I think `recenterButtonVisible` can be eliminated by checking if the `SummaryBottomSheet` is visible here instead with `view.isSummaryBottomSheetHidden()`.  We only want to show if the bottomsheet is currently hidden right?

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java [hunk 2]
```diff
@@ -316,6 +316,7 @@ public void hideInstructionList() {
    */
   public void showInstructionList() {
     onInstructionListVisibilityChanged(true);
+    instructionLayout.requestFocus();
```
Comment (danesfeder) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java]: Does the `MapView` regain focus okay after the list is hidden?

Comment (devotaaabel) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java]: Yup it does. I originally added `abandonFocus` to `hideInstructionList`, I can re-add it if you want

Comment (danesfeder) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java]: No problem, as long as you didn't see any issues

**Pull request merged**